### PR TITLE
[ADD] document_reindex

### PR DESCRIPTION
**Pull request by hbrunn**
# Reindex documents

This module allows you to reindex documents in case they were uploaded when the right configuration for indexation was missing.
# Usage

To reindex a single document, open its form and click the `Reindex document` button.

To reindex all documents, go to Settings / Configuration / Knowledge, check the box `Reindex all documents` or `Reindex all unindexed documents` and click apply. Those are done in the background, so watch your logs for the process to finish.

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Very useful tool! Only a minor aesthetic comment: put the arrow in white with a black border (or the contrary) for better contrast.

File: document_reindex/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    "name": "Reindex documents",
+    "version": "1.0",
+    "author": "Therp BV",
```
Comment (gurneyalex) [inline document_reindex/__openerp__.py]: could you add "Odoo Community Association (OCA)" ?

**Comment by gurneyalex** [comment]
Looks nice. A small automated test would be very welcome. 

Question: did you test how this behaves when you have a large set of (large) documents to reindex, especially regarding memory consumption?

**Comment by hbrunn** [comment]
@gurneyalex no, not yet. It probably needs something like https://github.com/OCA/knowledge/blob/8.0/attachments_to_filesystem/models/ir_attachment.py#L94

**Comment by hbrunn** [comment]
@pedrobaeza graphics is not excatly my field of excellence, but I tried...

**Comment by pedrobaeza** [comment]
Well, try it and if you are not able, I'll make it.

**Comment by hbrunn** [comment]
@pedrobaeza it's already commited. not beautiful, but it's mine as the saying goes. feel free to improve it if you've time

**Comment by pedrobaeza** [comment]
I think it can work. Thanks.

:+1:

**Comment by anthony-muschang** [comment]
Code review :+1:

**Comment by hbrunn** [comment]
@gurneyalex now I know, it crashes. 737469b fixes this.
The test is yet to come some time

**Comment by hbrunn** [comment]
@gurneyalex +test

**Comment by max3903** [comment]
:+1:

**Pull request merged**